### DISCUSSION TITLE
How to set a different terminal profile for debugpy in VSCode for macOS?

### DISCUSSION THREAD

**Discussion by gernophil** [microsoft/debugpy#1385] [Q&A]
I've seen [this post](https://www.reddit.com/r/vscode/comments/z5qt2c/possible_to_use_nondefault_terminal_profile_for/), but I don't know how this could help me. What I want to do is to start the debugger within a Rosetta terminal. What I currently do, is start the script in debug mode till it fails and then I type `arch -x86_64 zsh` to enter Rosetta2 emulation. Then I restart the debugger and then it works. How can I tell debugpy to immediately start with a Rosetta2 emulation?

I have added this to my `settings.json`:
```
    "terminal.integrated.profiles.osx": {
        "x86_64 zsh": {
            "path": "arch",
            "args": [
                "-arch", "x86_64", "zsh"
            ]
        }
    }
```
and I can use this as a normal terminal. But how do I select this for debugpy?

For debugpy within VSCode I switch to the `Run and debug` view an create a `launch.json` as shown below:
```
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Python: appName.py",
            "type": "python",
            "request": "launch",
            "program": "/abs/path/to/appName.py",
            "console": "integratedTerminal",
            "justMyCode": true
        }
    ]
}
```
Then I hit the start debugging button as shown in this screenshot:
https://code.visualstudio.com/assets/docs/python/debugging/debug-start-button.png
After that a debug terminal opens and fails since it is running on `arm64`. If I manually switch this terminal to `x86_64` and hit the run and debug button again, it works.

**Comment by r3m0t** [comment]
Setting `terminal.integrated.automationProfile.osx` ?

**Reply by gernophil** [reply]
But this I need to configure in the `settings.json` wouldn't it? I don't want this globally I want to have one `launch.json` to start the file with x86_64 and another one to debug the same file with arm64.

**Reply by gernophil** [reply]
I tried
```
    "settings": {
        "terminal.integrated.automationProfile.osx": {
            "path": "arch",
            "args": [
                "-x86_64", "zsh"
            ]
        }
    }
```
However, it still starts with `arm64` terminal. The same happens, if I put it directly in the `settings.json`.

**Comment by gernophil** [comment]
So, no one has an answer to this?

**Comment by int19h** [comment]
I'm not entirely sure how exactly VSCode matches terminals to profiles, but if it does so by terminal title, you could try setting `"consoleTitle": "x86_64 zsh"` in your launch.json.

**Reply by gernophil** [reply]
Unfortunately, I does not recognize this parameter in the `launch.json`. I should post this in the VSCode discussions maybe :).

**Reply by int19h** [reply]
If you mean that it gets a yellow squiggle in the editor, that's just VSCode validating against its JSON schema, which might not include this property. It should not preclude you from running it, though, and debugpy should pick it up. If a property is actually unsupported, you'll get an error message when running.

**Reply by gernophil** [reply]
Git it. Tried it today (had some other issues with VSCode yesterday), but still it starts in my deault `arm64` zsh. Just the name of the terminal window has changed to `x86_64 zsh`. To show you the error, I get this popup:
```
Exception: ImportError
dlopen(/Users/username/path/to/venv/lib/python3.9/site-packages/PIL/_imaging.cpython-39-darwin.so, 0x0002): tried: '/Users/username/path/to/venv/lib/python3.9/site-packages/PIL/_imaging.cpython-39-darwin.so' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/Users/username/path/to/venv/lib/python3.9/site-packages/PIL/_imaging.cpython-39-darwin.so' (no such file), '/Users/username/path/to/venv/lib/python3.9/site-packages/PIL/_imaging.cpython-39-darwin.so' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64'))
  File "/Users/username/Library/CloudStorage/OneDrive-Persönlich/playground/Git/noScribe/noScribe.py", line 24, in <module>
    from PIL import Image
ImportError: dlopen(/Users/username/path/to/venv/lib/python3.9/site-packages/PIL/_imaging.cpython-39-darwin.so, 0x0002): tried: '/Users/username/path/to/venv/lib/python3.9/site-packages/PIL/_imaging.cpython-39-darwin.so' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/Users/username/path/to/venv/lib/python3.9/site-packages/PIL/_imaging.cpython-39-darwin.so' (no such file), '/Users/username/path/to/venv/lib/python3.9/site-packages/PIL/_imaging.cpython-39-darwin.so' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64'))
```
I cancel this using ctrl+c, type `arch -x86_64 zsh` manually and run it again and then it works.

**Reply by int19h** [reply]
Ah well, it was worth a try. Unfortunately, that property is the only thing that I know of that debugpy can control when requesting a terminal from VSCode (we use the DAP `runInTerminal` request for this).
